### PR TITLE
Added ability get defaults index settings

### PR DESCRIPTION
**Pull request by krasilnikovm**

### CODE REVIEW

**Review by deguif — COMMENTED** [review]
@krasilnikovm thanks for this contribution.
Could you add a test to cover this new feature?

**Comment by krasilnikovm** [comment]
@deguif  I'll add

**Comment by krasilnikovm** [comment]
@deguif I added a tests, please take a look

**Comment by ruflin** [comment]
@krasilnikovm Could you add a changelog entry. I did approve the CI run. Thanks for your contribution!

File: tests/Index/SettingsTest.php [hunk 1]
```diff
@@ -30,11 +30,61 @@ public function testGet(): void
...
+        $client = $this->_getClient();
+        $index = $client->getIndex($indexName);
+        $index->create([], [
+            'recreate' => true,
+        ]);
+        $index->refresh();
+        $settings = $index->getSettings();
+
+        $this->assertIsArray($settings->get());
+        $this->assertEquals($settings->get('max_result_window', true), 10000);
```
Comment (deguif) [inline tests/Index/SettingsTest.php:55]:
```suggestion
        $this->assertEquals(10000, $settings->get('max_result_window', true));
```

Order for `assertEquals` is expected then actual ;)

File: tests/Index/SettingsTest.php [hunk 2]
```diff
@@ -30,11 +30,61 @@ public function testGet(): void
...
+                'max_result_window' => 100,
+            ],
+        ], [
+            'recreate' => true,
+        ]);
+
+        $index->refresh();
+        $settings = $index->getSettings();
+        $this->assertIsArray($settings->get());
+        $this->assertEquals($settings->get('max_result_window', true), 100);
```
Comment (deguif) [inline tests/Index/SettingsTest.php:82]: Same here

Comment (krasilnikovm) [reply tests/Index/SettingsTest.php]: My bad, fixed

File: tests/Index/SettingsTest.php (hunk 1 above)
Comment (krasilnikovm) [reply tests/Index/SettingsTest.php]: My bad, fixed

**Review by deguif — APPROVED** [review]

File: src/Index/Settings.php [hunk 3]
```diff
@@ -71,16 +71,24 @@ public function __construct(BaseIndex $index)
      *
      * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html
      */
-    public function get(string $setting = '')
+    public function get(string $setting = '', bool $includeDefaults = false)
     {
-        $requestData = $this->request()->getData();
+        $queryParameters = [
+            'include_defaults' => $includeDefaults,
+        ];
```
Comment (ruflin) [inline src/Index/Settings.php:78]: Surprised by this codecov comment as I think this part is covered in the tests. Maybe codecov is confused by the ] 🤷

**Review by ruflin — APPROVED** [review]

**Pull request merged**